op-chain-ops/crossdomain/types.go

```diff
@@ -30,3 +33,35 @@ type WithdrawalMessage interface {
 	Hash() (common.Hash, error)
 	StorageSlot() (common.Hash, error)
 }
+
+// InvalidMessage represents a message to the L1 message passer that
+// cannot be decoded as a withdrawal. They are defined as a separate
+// type in order to completely disambiguate them from any other
+// message.
+type InvalidMessage SentMessage
+
```

Comment:
Confirming that this serialization is correct based on the following code: https://github.com/ethereum-optimism/optimism/blob/6f1d8d9ab13d9cb1ef68360f74f82139152ac05d/packages/contracts/contracts/L2/predeploys/OVM_L2ToL1MessagePasser.sol#LL33C54-L33C54